src/routes/interactionsByIndex.mjs

```diff
@@ -0,0 +1,72 @@
+const MAX_INTERACTIONS_PER_PAGE = 1000;
+
+export const interactions = async (ctx) => {
+    const { page, limit, ownerAddress, indexes, contractTxId } = ctx.query;
+    const nodeDb = ctx.nodeDb;
+
+    const parsedPage = page ? parseInt(page) : 1
+    const parsedLimit = limit
+        ? Math.min(parseInt(limit), MAX_INTERACTIONS_PER_PAGE)
+        : MAX_INTERACTIONS_PER_PAGE;
+    const offset = parsedPage ? (parsedPage - 1) * parsedLimit : 0;
+
+    const bindings = [];
+    if (ownerAddress) {
+        bindings.push(ownerAddress);
+    }
+    if (contractTxId) {
+        bindings.push(contractTxId);
+    }
+
+    try {
+        const parsedIndexes = parseIndexes(indexes);
+        let indexQuery = `WHERE ? IN (tag_index_0, tag_index_1, tag_index_2, tag_index_3, tag_index_4)`;
+        bindings.push(parsedIndexes.shift())
+
+        for (const index of parsedIndexes) {
+            indexQuery += `AND ? IN (tag_index_0, tag_index_1, tag_index_2, tag_index_3, tag_index_4)`
```

Comment:
not sure I get it..why the 'IN' clause is generated twice?

Reply:
If someone pass `indexes=a` there is only one WHERE statement,
but if someone pass `indexes=a;b;c` it has to be executed multiple times cause we want to return interactions which has index `a` AND `b` AND `c` (so three times in this case). And in every step index can be in one of `tag_index_` columns.